tracer_test.go

```diff
@@ -808,3 +810,49 @@ func TestLocalEndpoint(t *testing.T) {
 		t.Errorf("IPv6 endpoint want %+v, have %+v", want.IPv6, have.IPv6)
 	}
 }
+
+func TestFinishedSpanHandlerAvoidsReporting(t *testing.T) {
+	rep := recorder.NewReporter()
+	defer rep.Close()
+
+	tracer, _ := NewTracer(
+		rep,
+		WithNoopSpan(false),
+		WithSampler(AlwaysSample),
+		WithFinishedSpanHandler(func(s *model.SpanModel) bool {
+			return false
+		}),
+	)
+	sp := tracer.StartSpan("test")
+	sp.Finish()
+
+	if want, have := 0, len(rep.Flush()); want != have {
+		t.Errorf("unexpected number of spans, want: %d, have: %d", want, have)
+	}
+}
+
+func TestFinishedSpanAddsTagsToSpan(t *testing.T) {
+	rep := recorder.NewReporter()
+	defer rep.Close()
+
+	tracer, _ := NewTracer(
+		rep,
+		WithNoopSpan(false),
+		WithSampler(AlwaysSample),
```

Comment:
need a test for never sample, that if the finished span handler needs all data it can still see it.

note in brave we have a flag on finished span handler which is always record or not (sampledLocal) you might need this to properly implement finished span handler.